src/game/GameScreen.js

```diff
@@ -22,29 +33,43 @@ export default class GameScreen extends React.Component {
     };
 
     render() {
+        if (this.state.game.idGame == 0) showWarning();
         return (
             <View style={styles.container}>
-                <Text>Quarto Android</Text>
-                <Text>This is the game</Text>
-                <Button
-                    style={styles.button}
-                    onPress={this.backHome}
-                    title="Back to home"
+                <Grid
+                    onPress={this.handleGridPress}
+                    grid={this.state.game.grid}
+                    readOnly={false}
+                />
+                <Text>Choose a piece for opponent</Text>
+                <RemainingList
+                    onPress={this.handleRemainingListPress}
+                    list={this.state.game.allPieces}
+                    readOnly={false}
                 />
             </View>
         );
     }
 
-    backHome = async () => {
+    handleGridPress = async (x, y) => {
+        try {
+            var newGame = await placePiece(this.state.game, x, y);
+            this.setState({
+                game: newGame,
+            });
+        } catch (error) {
+            showWarning();
```

Comment:
It would be a good idea to pass error details to warning

Reply:
why not